src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Checkout.php

```diff
@@ -78,20 +76,26 @@ public function logoutCustomer()
     }
 
     /**
-     * @param int        $resourceId
-     * @param int | null $customerId
+     * @param int          $resourceId
+     * @param int | string $storeId
+     * @param int | null   $customerId
      *
      * @return Shopgate_Cloudapi_Model_Auth_Code
+     * @throws Mage_Core_Exception
+     * @throws Mage_Core_Model_Store_Exception
      */
-    public function generateAuthToken($resourceId, $customerId)
+    public function generateAuthToken($resourceId, $storeId, $customerId)
     {
         /** @var Magento_Db_Adapter_Pdo_Mysql $writeConnection */
         $writeConnection = Mage::getSingleton('core/resource')->getConnection('core_write');
         /** @var Shopgate_Cloudapi_Model_OAuth2_Db_Pdo $storage */
         $storage = Mage::getModel('shopgate_cloudapi/oAuth2_db_pdo', array($writeConnection->getConnection()));
-        $token   = $storage->createAuthItemByType(
+        $storage->setStore(Mage::app()->getStore($storeId));
+
```

Comment:
If my assumption from the 1st comment is correct, $storeId should already be the store instance, not need to fetch it again using getStore().